File: transport_marchandises_web/src/repository/ReservationRepository.js
import ApiService from "../service/ApiService";
import ApiRoutes from "../routes/ApiRoutes.js";

class ReservationRepository {
  constructor(apiService = ApiService) {
    this.apiService = apiService;
    this.pathReservation = ApiRoutes.reservation;
    this.paramsVehicule = "?vehicule_id=";
  }

  // ➕ Ajouter une réservation
  async postReservation(id, data) {
    const dataResponse = await this.apiService.post(
      `${this.pathReservation}${this.paramsVehicule}${id}`,
      data
    );
    console.log
    return dataResponse;
  }

  // 📋 Récupérer les réservations d’un véhicule
  async getReservationVehicule(id) {
    const data = await this.apiService.get(
      `${this.pathReservation}${this.paramsVehicule}${id}`
    );
    return data;
  }

  // ✏️ Modifier une réservation
  async patchReservation(id, data = {}) {
    const dataResponse = await this.apiService.patch(
      `${this.pathReservation}${id}/`,
      data
    );
    return dataResponse;
  }

  // ❌ Supprimer une réservation
  async deleteReservation(id) {
    const data = await this.apiService.delete(`${this.pathReservation}${id}/`);
    return data;
  }
}
export default new ReservationRepository();

File: transport_marchandises/lib/Repository/reservation_repository.dart
import '../Service/Service.dart';
import '../routes/routes.dart';

class ReservationRepository {
  final ApiService apiService;
  ReservationRepository(this.apiService);
  final String pathReservation = Routes.resirvation;
  final String paramsVecule = "?vehicule_id=";
  // add Resirvation
  Future<Map<String, dynamic>> postReservation(
    id,
    Map<String, dynamic> data,
  ) async {
    final dataResponse = await apiService.post(
      "${pathReservation}/${paramsVecule}${id}",
      data: data,
    );
    return dataResponse;
  }

  // get les reservation
  Future<List<dynamic>?> getReservationVehicule(id) async {
    final data = await apiService.get(
      "${pathReservation}/${paramsVecule}${id}",
    );
    // print(data.runtimeType);
    return data;
  }

  Future<Map<String, dynamic>> patchVehicule(
    int id, {
    Map<String, dynamic>? data,
  }) async {
    final dataResponse = await apiService.patch(
      "${pathReservation}/${id}/",
      data: data,
    );
    return dataResponse;
  }

  Future<dynamic> delete(int id) async {
    final data = await apiService.delete("${pathReservation}/${id}/");
    return data;
  }
}
